interchaintest/misbehaviour_test.go

```diff
@@ -0,0 +1,315 @@
+	)
+
+	chains, err := cf.Chains(t.Name())
+	require.NoError(t, err)
+	chainA, chainB := chains[0].(*cosmos.CosmosChain), chains[1].(*cosmos.CosmosChain)
+
+	ctx := context.Background()
+	client, network := interchaintest.DockerSetup(t)
+
+	rf := relayertest.NewRelayerFactory(relayertest.RelayerConfig{InitialBlockHistory: 50})
+	r := rf.Build(t, client, network)
+
+	const pathChainAChainB = "chainA-chainB"
+
+	ic := interchaintest.NewInterchain().
+		AddChain(chainA).
+		AddChain(chainB).
+		AddRelayer(r, "relayer").
+		AddLink(interchaintest.InterchainLink{
+			Chain1:           chainA,
+			Chain2:           chainB,
+			Relayer:          r,
+			Path:             pathChainAChainB,
+			CreateClientOpts: ibc.CreateClientOptions{TrustingPeriod: "15m"},
+		})
+
+	rep := testreporter.NewNopReporter()
+	eRep := rep.RelayerExecReporter(t)
+
+	require.NoError(t, ic.Build(ctx, eRep, interchaintest.InterchainBuildOptions{
+		TestName:         t.Name(),
+		Client:           client,
+		NetworkID:        network,
+		SkipPathCreation: false,
+	}))
+
+	t.Cleanup(func() {
+		_ = ic.Close()
+	})
+
+	// create a new user account and wait a few blocks for it to be created on chain
+	user := interchaintest.GetAndFundTestUsers(t, ctx, "user-1", 10_000_000, chainA)[0]
+	err = testutil.WaitForBlocks(ctx, 5, chainA)
+
+	// Start the relayer
+	require.NoError(t, r.StartRelayer(ctx, eRep, pathChainAChainB))
+
+	t.Cleanup(
+		func() {
+			err := r.StopRelayer(ctx, eRep)
+			if err != nil {
+				panic(fmt.Errorf("an error occured while stopping the relayer: %s", err))
+			}
+		},
+	)
+
+	// query latest height on chainB
+	latestHeight, err := chainB.Height(ctx)
+	require.NoError(t, err)
+
+	// query header at height on chainB
+	h := int64(latestHeight)
+	header, err := queryHeaderAtHeight(ctx, t, h, chainB)
+	require.NoError(t, err)
+
+	// query tm client state on chainA
+	const clientID = "07-tendermint-0"
+
+	cmd := []string{"ibc", "client", "state", clientID}
+
+	stdout, stderr, err := chainA.Validators[0].ExecQuery(ctx, cmd...)
+	require.NoError(t, err)
+	require.Empty(t, stderr)
+
+	queryResp := clienttypes.QueryClientStateResponse{}
+	err = defaultEncoding().Codec.UnmarshalJSON(stdout, &queryResp)
+	require.NoError(t, err)
+
+	clientState, err := clienttypes.UnpackClientState(queryResp.ClientState)
+	require.NoError(t, err)
+
+	// get latest height from prev client state above & create new height + 1
+	height := clientState.GetLatestHeight().(clienttypes.Height)
+	newHeight := clienttypes.NewHeight(height.RevisionNumber, height.RevisionHeight+1)
+
+	// create a validator for signing duplicate header
+	keyBz, err := chainB.Validators[0].ReadFile(ctx, "config/priv_validator_key.json")
+	require.NoError(t, err)
+
+	pvk := cosmos.PrivValidatorKeyFile{}
+	err = json.Unmarshal(keyBz, &pvk)
+	require.NoError(t, err)
+
+	decodedKeyBz, err := base64url.Decode(pvk.PrivKey.Value)
```

Comment:
Sorry I missed this last time, but we can avoid the extra library import here by importing `encoding/base64` and using `base64.StdEncoding.DecodeString`

Reply:
Ahh this was some IDE magic that I didn't notice either, I was trying to use the stdlib `base64` package. I'll get this taken care of.